src/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { PropTypes } from 'prop-types';
+import classNames from 'classnames';
+import { makeEmojiArray, makePagesArray } from '../helpers/makePagesArray';
+
+export const Pagination = ({
+  pagin,
+  onChangePage,
+  onClickPrev,
+  onClickNext,
+  onChangePerPage,
+  onClickOption,
+}) => {
+  const { total, perPage, page, option } = pagin;
+  const lastPage = Math.ceil(total / perPage);
+  const forPage = ((page - 1) * perPage);
+  const toPage = perPage * page > total
+    ? total : perPage * page;
+  const location = useLocation();
+  const history = useHistory();
+  const searchParams = new URLSearchParams(location.search);
+  const changeHistory = (name, value) => {
+    searchParams.set(name, value);
+    history.push(`?${searchParams.toString()}`);
+  };
+
+  return (
+    <div>
+      <div className="emoji">
+        {makeEmojiArray(total).slice(forPage, toPage).map(t => (
+          <span className="emoji" key={t}>{t}</span>
+        ))}
+      </div>
+      <div>
+        <span>
+          {`${forPage + 1} - ${toPage} of ${total}`}
+        </span>
+        <select
+          typy="select"
+          value={perPage}
+          onChange={(e) => {
+            onChangePerPage(e);
+            changeHistory('page', 1);
+            changeHistory('perPage', e.target.value);
+          }}
+        >
+          {[3, 5, 10, 20].map(item => (
+            <option key={item} value={item}>{item}</option>
+          ))}
+        </select>
+        <button
+          type="button"
+          className={classNames('option', {
+            paginFerst: option,
+            paginSecond: !option,
```

Comment:
не дуже розумію для чого ці класи, вони якісь дуже специфічні, навряд вони насправді треба, з назви не зрозуміло для чого вони
```suggestion
            paginationFirst: option,
            paginationSecond: !option,
```